src/Node/Net.purs

```diff
@@ -0,0 +1,585 @@
+module Node.Net
+  ( Address
+  , ConnectOptions
+  , ListenOptions
+  , Lookup
+  , Server
+  , ServerOptions
+  , Socket
+  , SocketOptions
+  , address
+  , allowHalfOpen
+  , backlog
+  , bufferSize
+  , bytesRead
+  , bytesWritten
+  , close
+  , connect
+  , connectICP
+  , connectTCP
+  , connecting
+  , createConnection
+  , createConnectionICP
+  , createConnectionTCP
+  , createServer
+  , destroy
+  , destroyed
+  , end
+  , endString
+  , exclusive
+  , family
+  , fd
+  , getConnections
+  , hints
+  , host
+  , ipv6Only
+  , isIP
+  , isIPv4
+  , isIPv6
+  , listen
+  , listenICP
+  , listenTCP
+  , listening
+  , localAddress
+  , localAddressOption
+  , localPort
+  , localPortOption
+  , onCloseServer
```

Comment:
`onCloseServer` and `onCloseSocket` are suffixed because both the `Server` and the `Socket` have `'close'` events. Are these suffixes okay? Should we have separate modules: `Node.Net.Server`, `Node.Net.Socket` and each have an `onClose`? Should we do something else?

Reply:
I think having a `Node.Net.Server` and a `Node.Net.Socket` module sounds good; this module is quite big as-is.